### PR TITLE
Update quote API to use Dad Jokes

### PR DESCRIPTION
**Pull request by oryolo**

### CODE REVIEW

File: .github/actions/slack-action/src/Quote.ts [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+import { get, Header, HttpError, toHttpTask } from 'ajaxios';
 import Task from 'taskarian';
-import { quoteDecoder } from './Decoders';
-import { ZenQuote } from './Types';
+import { jokeDecoder } from './Decoders';
+import { Joke } from './Types';
 
-export const getZenQuote = (): Task<HttpError, ZenQuote> =>
-  toHttpTask(get('https://zenquotes.io/api/random').withDecoder(quoteDecoder));
+const href = 'https://icanhazdadjoke.com/';
```
Comment (costajohnt) [inline .github/actions/slack-action/src/Quote.ts:6]: maybe we can use cooperts url package to verify this url `toUrlT`?

File: .github/actions/slack-action/package.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
     "@execonline-inc/environment": "workspace:*",
     "@execonline-inc/error-handling": "workspace:*",
     "@execonline-inc/logging": "workspace:*",
-    "@execonline-inc/maybe-adapter": "^5.0.0",
```
Comment (costajohnt) [inline .github/actions/slack-action/package.json:13]: did maybe adapter get removed? i thought we only made additive changes here

File: .pnp.cjs
Comment (costajohnt) [inline .pnp.cjs:1]: not sure we need this file, can we try removing?

File: .github/actions/slack-action/src/Quote.ts [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+const parseUrlTask = (href: string): Task<QuoteError, URLParser> => toUrlT(href);
+
+export const getJoke = (url: URLParser): Task<QuoteError, Joke> =>
+  toHttpTask(
+    get(url.toString())
+      .withHeader(header)
+      .withDecoder(jokeDecoder),
+  );
+
+export const runJoke = () => parseUrlTask(href).andThen(getJoke);
```
Comment (costajohnt) [inline .github/actions/slack-action/src/Quote.ts:22]: I think we could do a bit better with some of the naming
```
export const fetchDadJoke = () => validateUrl(href).andThen(getJoke);
```

Comment (oryolo) [reply .github/actions/slack-action/src/Quote.ts]: I agree. Done!

**Review by costajohnt — APPROVED** [review]

**Pull request merged**